test/integration/lockfile/test_user_overrides.py

```diff
@@ -360,3 +360,78 @@ def test_lock_update(self, kind, old, new):
         lock = c.load("conan.lock")
         assert old not in lock
         assert new in lock
+
+
+
+class TestLockUpgrade:
+    @pytest.mark.parametrize("kind, pkg, old, new", [
+        ("requires", "math", "math/1.0", "math/1.1"),
+        ("build-requires", "cmake", "cmake/1.0", "cmake/1.1"),     # TODO there is not a --build-requires
+        # ("python-requires", "mytool", "mytool/1.0", "mytool/1.1"), # TODO nor a --python-requires
+    ])
+    def test_lock_upgrade(self, kind, pkg, old, new):
+        c = TestClient(light=True)
+        c.save({f"{pkg}/conanfile.py": GenConanfile(pkg)})
+
+        c.run(f"export {pkg} --version=1.0")
+        rev0 = c.exported_recipe_revision()
```

Comment:
You don't need the revisions check if you are upgrading the versions, test can be simplified